server/api.go

```diff
@@ -67,6 +76,83 @@ func (a *API) getAvatar(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (a *API) decryptEncryptedContentData(key []byte, encryptedContent msteams.EncryptedContent) ([]byte, error) {
+	ciphertext, err := base64.StdEncoding.DecodeString(encryptedContent.Data)
+	if err != nil {
+		a.p.API.LogDebug("Unable to decode encrypted data", "error", err)
+		return nil, err
+	}
+	msDataSignature, err := base64.StdEncoding.DecodeString(encryptedContent.DataSignature)
+	if err != nil {
+		a.p.API.LogDebug("Unable to decode data signature", "error", err)
+		return nil, err
+	}
+
+	mac := hmac.New(sha256.New, key)
+	mac.Write(ciphertext)
+	expectedMac := mac.Sum(nil)
+	if !hmac.Equal(expectedMac, msDataSignature) {
+		a.p.API.LogDebug("Invalid data signature", "error", errors.New("The key signature doesn't match"))
+		return nil, errors.New("The key signature doesn't match")
+	}
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ciphertext) < aes.BlockSize {
+		return nil, errors.New("ciphertext too short")
+	}
+	iv := key[:16]
+
+	if len(ciphertext)%block.BlockSize() != 0 {
+		return nil, errors.New("ciphertext is not a multiple of the block size")
+	}
+
+	mode := cipher.NewCBCDecrypter(block, iv)
+	result := make([]byte, len(ciphertext))
+	mode.CryptBlocks(result, ciphertext)
+	resultPadding := int(result[len(result)-1])
+	result = result[:len(result)-resultPadding]
+	return result, nil
+}
+
+func (a *API) decryptEncryptedContentDataKey(encryptedContent msteams.EncryptedContent) ([]byte, error) {
+	ciphertext, err := base64.StdEncoding.DecodeString(encryptedContent.DataKey)
+	if err != nil {
+		a.p.API.LogDebug("Unable to decode key", "error", err)
+		return nil, err
+	}
+
+	key, err := a.p.getPrivateKey()
+	if err != nil {
+		a.p.API.LogDebug("Unable to get private key", "error", err)
+		return nil, err
+	}
+	hash := sha1.New() //nolint:gosec
```

Comment:
Let's use sha256 instead of sha1. I think that's supported in MS with the naming `RSA-OAEP-256`: https://learn.microsoft.com/en-us/azure/key-vault/keys/about-keys-details. MS doc on how to decrypt the data key only mentioned OAEP without specifying the hash, so I hope that sha256 is supported there too: https://learn.microsoft.com/en-us/graph/webhooks-with-resource-data#decrypting-resource-data